src/user/user.service.ts

```diff
@@ -103,6 +103,27 @@ export class UserService {
     return elderly;
   }
 
+  async checkRelationship(eid: number, cid: number) {
+    let elderly;
+    let isRelated = false;
+    try {
+      elderly = await this.findElderlyById(eid);
+    } catch {
+      throw new NotFoundException('Elderly Does Not Exist');
+    }
+      
+    const caretakers = await this.findCaretakerByElderlyId(elderly.uid);
+    if (!caretakers) return isRelated;
```

Comment:
actually I think this service should send error when the caretaker is not relate to elderly. So my suggestion is have the error handling here and if the caretaker is related to elderly then just send true back.(But there is no need for using that boolean that u send back so in the controller u can just write 
await this.userService.checkRelationship(elderlyId, caretakerId);